From 184cd73c7c1b8dce6f7bb97b1897b3b698208998 Mon Sep 17 00:00:00 2001
From: Alex Cruz <du.alexcruz@gmail.com>
Date: Fri, 10 Nov 2017 20:41:55 -0500
Subject: [PATCH 077/296] Fix NPE when AOSP ambient display preference is
 removed

 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method
 'void android.support.v7.preference.Preference.setOnPreferenceChangeListener(android.support.v7.preference.Preference$OnPreferenceChangeListener)'  on a null object reference
 at com.android.settingslib.core.AbstractPreferenceController.displayPreference(AbstractPreferenceController.java:27)
 at com.android.settings.dashboard.DashboardFragment.displayResourceTiles(DashboardFragment.java:239)
 at com.android.settings.dashboard.DashboardFragment.refreshAllPreferences(DashboardFragment.java:277)
---
 .../com/android/settingslib/core/AbstractPreferenceController.java  | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java b/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
index 38fe8790e4d..e6f82b8556d 100644
--- a/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
+++ b/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
@@ -23,8 +23,10 @@ public abstract class AbstractPreferenceController {
       if (isAvailable()) {
           if (this instanceof Preference.OnPreferenceChangeListener) {
               final Preference preference = screen.findPreference(getPreferenceKey());
-              preference.setOnPreferenceChangeListener(
-                      (Preference.OnPreferenceChangeListener) this);
+                if (preference != null) {
+                    preference.setOnPreferenceChangeListener(
+                            (Preference.OnPreferenceChangeListener) this);
+                }
           }
       } else {
           removePreference(screen, getPreferenceKey());
-- 
2.11.0

